.github/workflows/deploy-branch.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -eux
+
+COMPONENT=${1:?"Require components as first argument, e.g. frontend or workspace-service"}
+
+# Checkout correct branch in the deployments repo
+cd $GITHUB_WORKSPACE/futurenhs-deployments
+git config --local user.email "futurenhs-devs@red-badger.com"
+git config --local user.name "FutureNHS CI/CD"
+if git fetch --unshallow origin pr-$GITHUB_HEAD_REF master; then
+	git checkout origin/pr-$GITHUB_HEAD_REF -b pr-$GITHUB_HEAD_REF
+	git merge origin/master --strategy=recursive --strategy-option=ours
+else
+	git checkout HEAD -b pr-$GITHUB_HEAD_REF
+fi
+
+# Install Kustomize
+cd $HOME
+curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash
+
+# Prepare manifests and copy to deployments repo
+if [ "$COMPONENT" = "infrastructure" ]; then
+	$GITHUB_WORKSPACE/futurenhs-platform/infrastructure/scripts/create-dev-overlays.py
+	mkdir -p $GITHUB_WORKSPACE/futurenhs-deployments/cert-manager
+	cp -r $GITHUB_WORKSPACE/futurenhs-platform/infrastructure/kubernetes/cert-manager/* $GITHUB_WORKSPACE/futurenhs-deployments/cert-manager
+	mkdir -p $GITHUB_WORKSPACE/futurenhs-deployments/ingress
+	cp -r $GITHUB_WORKSPACE/futurenhs-platform/infrastructure/kubernetes/ingress/* $GITHUB_WORKSPACE/futurenhs-deployments/ingress
+else
+	cd $GITHUB_WORKSPACE/futurenhs-platform/$COMPONENT/manifests/dev-template
+	$HOME/kustomize edit set image $DIGEST
+	$GITHUB_WORKSPACE/futurenhs-platform/infrastructure/scripts/create-dev-overlays.py
+	mkdir -p $GITHUB_WORKSPACE/futurenhs-deployments/$COMPONENT
+	cp -r $GITHUB_WORKSPACE/futurenhs-platform/$COMPONENT/manifests/* $GITHUB_WORKSPACE/futurenhs-deployments/$COMPONENT
+fi
+
+# Get commit message
+cd $GITHUB_WORKSPACE/futurenhs-platform
+git fetch origin HEAD --deepen=2
+COMMIT_MESSAGE=$(git log --no-merges -1 --format=%s)
```

Comment:
Thinking about it, I think that something like this might fix the commit messages:
```suggestion
cd $GITHUB_WORKSPACE/futurenhs-platform
git fetch origin $GITHUB_HEAD_REF --deepen=2
COMMIT_MESSAGE=$(git log origin/$GITHUB_HEAD_REF --no-merges -1 --format=%s)
```
(I haven't tried it out though, so might be best to do it in its own PR)

Reply:
The commit messages are actually fixed. I think I found the problem. We were using `${{ github.event.after }}` to refer to the commit. This properties does not exist in pull request events. Either it got removed at some point or maybe it only exists in push events. This new method works fine.

Reply:
As part of the build log you can see:

```
++ git log --no-merges -1 --format=%s
+ COMMIT_MESSAGE='Update master builds'
```

Reply:
Btw, you can see my attempts on the latest build on this PR: https://github.com/FutureNHS/futurenhs-platform/pull/463